kyuubi-common/src/main/scala/org/apache/kyuubi/KyuubiSQLException.scala

```diff
@@ -26,7 +26,20 @@ import scala.collection.JavaConverters._
 
 import org.apache.hive.service.rpc.thrift.{TStatus, TStatusCode}
 
-class KyuubiSQLException(msg: String, cause: Throwable) extends SQLException(msg, cause) {
+/**
+ * @param msg        a description of the exception
+ * @param sqlState   an XOPEN or SQL:2003 code identifying the exception
+ * @param errorCode  a database vendor-specific exception code
+ * @param cause      the underlying reason for this [[SQLException]]
+ *                   (which is saved for later retrieval by the `getCause()` method);
+ *                   may be null indicating the cause is non-existent or unknown.
+ */
+class KyuubiSQLException(msg: String, cause: Throwable, sqlState: String, errorCode: Int)
+  extends SQLException(msg, sqlState, errorCode, cause) {
+
+  // for reflection
```

Comment:
where do we use this?

Reply:
https://github.com/NetEase/kyuubi/blob/b9079b698d0239adad1fda945cb82f59c7eca950/kyuubi-common/src/main/scala/org/apache/kyuubi/KyuubiSQLException.scala#L116-L124